src/tests/cmocka/test_dp_req.c

```diff
@@ -0,0 +1,362 @@
+/*
+    Copyright (C) 2015 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <talloc.h>
+#include <tevent.h>
+#include <errno.h>
+#include <popt.h>
+
+#include "providers/backend.h"
+#include "providers/data_provider/dp_private.h"
+#include "tests/cmocka/common_mock.h"
+#include "tests/common.h"
+
+struct test_ctx {
+    TALLOC_CTX *mem_ctx;
+};
+
+static int test_setup(void **state)
+{
+    struct test_ctx *test_ctx = NULL;
+
+    test_ctx = talloc_zero(global_talloc_context, struct test_ctx);
+    assert_non_null(test_ctx);
+
+    test_ctx->mem_ctx = talloc_new(NULL);
+    assert_non_null(test_ctx->mem_ctx);
+
```

Comment:
You should call check_leaks_setup/teardown. You don't need to use test_ctx at all, just call check_leaks_setup here and use global_talloc_context in the tests.